### PR TITLE
TOSA to TTIR: adding simple op conversion patterns

### PR DESCRIPTION
**Pull request by sdjukicTT**
Added simple patterns that use the `TosaToTTIRDefaultDPSOpConversionPattern`. TOSA operations for which support is added:
- [tosa.cast](https://mlir.llvm.org/docs/Dialects/TOSA/#tosacast-mlirtosacastop)[ (mlir::tosa::CastOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosacast-mlirtosacastop);
- [tosa.ceil](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaceil-mlirtosaceilop)[ (mlir::tosa::CeilOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaceil-mlirtosaceilop);
- [tosa.cos](https://mlir.llvm.org/docs/Dialects/TOSA/#tosacos-mlirtosacosop)[ (mlir::tosa::CosOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosacos-mlirtosacosop);
- [tosa.equal](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaequal-mlirtosaequalop)[ (mlir::tosa::EqualOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaequal-mlirtosaequalop);
- [tosa.exp](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaexp-mlirtosaexpop)[ (mlir::tosa::ExpOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaexp-mlirtosaexpop);
- [tosa.floor](https://mlir.llvm.org/docs/Dialects/TOSA/#tosafloor-mlirtosafloorop)[ (mlir::tosa::FloorOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosafloor-mlirtosafloorop);
- [tosa.greater](https://mlir.llvm.org/docs/Dialects/TOSA/#tosagreater-mlirtosagreaterop)[ (mlir::tosa::GreaterOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosagreater-mlirtosagreaterop);
- [tosa.logical_and](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_and-mlirtosalogicalandop)[ (mlir::tosa::LogicalAndOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_and-mlirtosalogicalandop);
- [tosa.logical_not](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_not-mlirtosalogicalnotop)[ (mlir::tosa::LogicalNotOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_not-mlirtosalogicalnotop);
- [tosa.logical_or](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_or-mlirtosalogicalorop)[ (mlir::tosa::LogicalOrOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_or-mlirtosalogicalorop);
- [tosa.logical_xor](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_xor-mlirtosalogicalxorop)[ (mlir::tosa::LogicalXorOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosalogical_xor-mlirtosalogicalxorop).

Also added a test for each in `test/ttmlir/Conversion/TosaToTTIR`, as well as a test for:
-  [tosa.abs](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaabs-mlirtosaabsop)[ (mlir::tosa::AbsOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaabs-mlirtosaabsop);
- [tosa.add](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaadd-mlirtosaaddop)[ (mlir::tosa::AddOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosaadd-mlirtosaaddop);
- [tosa.greater_equal](https://mlir.llvm.org/docs/Dialects/TOSA/#tosagreater_equal-mlirtosagreaterequalop)[ (mlir::tosa::GreaterEqualOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosagreater_equal-mlirtosagreaterequalop).

Moved the test for [tosa.mul](https://mlir.llvm.org/docs/Dialects/TOSA/#tosamul-mlirtosamulop)[ (mlir::tosa::MulOp)](https://mlir.llvm.org/docs/Dialects/TOSA/#tosamul-mlirtosamulop) in the `test/ttmlir/Conversion/TosaToTTIR` folder as well.

closes https://github.com/tenstorrent/tt-mlir/issues/1391 closes https://github.com/tenstorrent/tt-mlir/issues/1367 closes https://github.com/tenstorrent/tt-mlir/issues/1382 closes https://github.com/tenstorrent/tt-mlir/issues/1392 closes https://github.com/tenstorrent/tt-mlir/issues/1441 closes https://github.com/tenstorrent/tt-mlir/issues/1442 closes https://github.com/tenstorrent/tt-mlir/issues/1393 closes https://github.com/tenstorrent/tt-mlir/issues/1395 closes https://github.com/tenstorrent/tt-mlir/issues/1396 closes https://github.com/tenstorrent/tt-mlir/issues/1397 closes https://github.com/tenstorrent/tt-mlir/issues/1394

### CODE REVIEW

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 1]
```diff
@@ -120,6 +150,8 @@ void populateTosaToTTIRPatterns(MLIRContext *ctx, RewritePatternSet &patterns,
                                 TypeConverter &typeConverter) {
   addElementwiseUnaryOpsConversionPatterns(ctx, patterns, typeConverter);
   addElementwiseBinaryOpsConversionPatterns(ctx, patterns, typeConverter);
+  addLogicalOpConversionPattern(ctx, patterns, typeConverter);
+  addLogicalOpConversionPattern(ctx, patterns, typeConverter);
```
Comment (ajakovljevicTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp:154]: You added this line two times, might be an oversight.

File: test/ttmlir/Conversion/TosaToTTIR/compare/equal.mlir [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// RUN: ttmlir-opt --convert-tosa-to-ttir %s | FileCheck %s
+module attributes {} {
+  func.func @test_equal(%arg0: tensor<13x21x3xf32>, %arg1: tensor<13x21x3xf32>) -> tensor<13x21x3xi1> {
+    %0 = tosa.equal %arg0, %arg1 : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xi1>
+    // CHECK: %[[EQ_OUT:[0-9]+]] = tensor.empty() : tensor<13x21x3xi1
+    // CHECK: %{{[0-9]+}} = "ttir.eq"(%arg{{[0-9]+}}, %arg{{[0-9]+}}, %[[EQ_OUT]]){{.+}} -> tensor<13x21x3xi1>
+    return %0 : tensor<13x21x3xi1>
```
Comment (ajakovljevicTT) [inline test/ttmlir/Conversion/TosaToTTIR/compare/equal.mlir:7]: Great job on the detailed tests! Could you also add the return check? 
Something like: // CHECK: return [[VAL1]] : [[TENSOR_SIZE]]

File: test/ttmlir/Conversion/TosaToTTIR/compare/equal.mlir [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// RUN: ttmlir-opt --convert-tosa-to-ttir %s | FileCheck %s
+module attributes {} {
+  func.func @test_equal(%arg0: tensor<13x21x3xf32>, %arg1: tensor<13x21x3xf32>) -> tensor<13x21x3xi1> {
+    %0 = tosa.equal %arg0, %arg1 : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xi1>
+    // CHECK: %[[EQ_OUT:[0-9]+]] = tensor.empty() : tensor<13x21x3xi1
```
Comment (ajakovljevicTT) [inline test/ttmlir/Conversion/TosaToTTIR/compare/equal.mlir:5]: missing '>' ad the end of the line

**Review by ajakovljevicTT — APPROVED** [review]

**Comment by kmitrovicTT** [comment]
@sdjukicTT you can do

closes https://github.com/tenstorrent/tt-mlir/issues/1391, https://github.com/tenstorrent/tt-mlir/issues/1367, https://github.com/tenstorrent/tt-mlir/issues/1382...

instead of

closes https://github.com/tenstorrent/tt-mlir/issues/1391 closes https://github.com/tenstorrent/tt-mlir/issues/1367 closes https://github.com/tenstorrent/tt-mlir/issues/1382...

**Pull request merged**